test/react/hooks/internal/use-chat-client-context.test.tsx

```
@@ -0,0 +1,185 @@
+};
+
+vi.mock('ably');
+
+describe('useChatClientContext', () => {
+  afterEach(() => {
+    cleanup();
+  });
+
+  it('should throw an error if used outside of ChatClientProvider', () => {
+    const TestThrowError: React.FC = () => {
+      expect(() => useChatClientContext()).toThrowErrorInfo({
```

Biome (2.1.2)

[error] This hook is being called from a nested function, but all hooks must be called unconditionally from the top-level component.

For React to preserve state between calls, hooks needs to be called unconditionally and always in the same order.
See https://reactjs.org/docs/hooks-rules.html#only-call-hooks-at-the-top-level

(lint/correctness/useHookAtTopLevel)